openfisca_core/entities.py

```diff
@@ -275,6 +275,44 @@ def value_from_partner(self, array, entity, role):
             [value_subrole_2, value_subrole_1],
             )
 
+    def get_rank(self, entity, criteria, condition = True):
+        """
+        Get the rank of a person within an entity according the a criteria.
+        The person with rank 0 has the minimum value of criteria
+        If condition is specified, then the persons who don't respect it are not taken into account.
+
+        Exemple:
+
+        >>> age = person('age', period)  # e.g [32, 34, 2, 8, 1]
+        >>> person.get_rank(household, age)
+        >>> [3, 4, 0, 2, 1]
+
+        >>> is_child = person.has_role(Household.CHILD)  # [False, False, True, True, True]
+        >>> person.get_rank(household, - age, condition = is_child)  # Sort in reverse order to get the elder child first
+        >>> [-1, -1, 1, 0, 2]
+        """
+
+        positions = entity.members_position
+        biggest_entity_size = np.max(positions) + 1
+        filtered_criteria = np.where(condition, criteria, np.inf)
+        ids = entity.members_entity_id
+
+        # Matrix: the value in line i and column j is the value of criteria for the jth person of the ith entity
+        matrix = np.asarray([
+            entity.value_kth_person(k, filtered_criteria, default = np.inf)
+            for k in range(biggest_entity_size)
+            ]).transpose()
+
+        # We double-argsort all lines of the matrix.
+        # Double-argsorting gets the rank of each value once sorted
```

Comment:
```py
x = np.asarray([3,1,6,4,0])
y = np.argsort(x)  # [4, 1, 0, 3, 2], because the 4th value (0) is the smallest, the 1st value (1) is the second smallest, the 0th value (3) is the third smallest one, etc.
# Note that in vector `y`, the ith value does *not* correspond to the ith entity.
```

Now with a second argsort:
```
z = np.argsort(y) # [2, 1, 4, 3, 0]
The smallest value (0) of y is the 2nd value. Given the definition of y, this means that the entity 0th entity was ranked #2 by order of x. 
The second smallest value (1) of y is the the 1st value. Similarly, this means that the 1st entity was ranked #1 by order of x. 
```

Reply:
I'm adding a brief comment to the code

Reply:
🥇